pkg/rehearse/jobs.go

```diff
@@ -138,17 +173,10 @@ func filterJob(source *prowconfig.Presubmit, allowVolumes bool) error {
 			return fmt.Errorf("cannot rehearse jobs that call ci-operator with '--git-ref' arg")
 		}
 	}
-	if len(source.Spec.Volumes) > 0 && !allowVolumes {
+	if len(spec.Volumes) > 0 && !allowVolumes {
 		return fmt.Errorf("jobs that need additional volumes mounted are not allowed")
 	}
 
-	if len(source.Branches) == 0 {
```

Comment:
If the periodic has `ExtraRefs` set, use the first item in that list as equivalent to a `Refs` field